tests/test_core.py

```diff
@@ -166,23 +167,44 @@ def test_years_active_extend(test_mp):
 
 def test_new_timeseries(test_mp):
     scen = test_mp.TimeSeries(*test_args, version='new', annotation='testing')
-    df = {'region': ['World'], 'variable': ['Testing'], 'unit': ['???'],
-          'year': [2020], 'value': [23.6]}
+    df = {'year': [2010, 2020], 'value': [23.5, 23.6]}
     df = pd.DataFrame.from_dict(df)
+    df['region'] = 'World'
+    df['variable'] = 'Testing'
+    df['unit'] = '???'
     scen.add_timeseries(df)
```

Comment:
I have used the same formulation as above df['xyz'] = 'abc' and always receive a warning from python 3.6 when doing this. An alternative way of writing this which seems to be safer according to the link below is df.loc[:,'XYZ'] = 'abc'
see below
"A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  df['unit'] = 'GWa'"

Reply:
I know this error message happens in some instances, but I don't get it here... I guess it's not a problem here because this is a new dataframe (as opposed to an dataframe cut from an existing one)?